src/idl_gen_dart.cpp

```diff
@@ -449,13 +449,22 @@ class DartGenerator : public BaseGenerator {
     code += "  final " + _kFb + ".BufferContext _bc;\n";
     code += "  final int _bcOffset;\n\n";
 
-    GenImplementationGetters(struct_def, &code);
+    std::vector<std::pair<int, FieldDef*>> non_deprecated_fields;
+    for (auto it = struct_def.fields.vec.begin();
+           it != struct_def.fields.vec.end(); ++it) {
+        auto &field = **it;
+        if (field.deprecated) continue;
+        auto offset = it - struct_def.fields.vec.begin();
+        non_deprecated_fields.push_back(std::make_pair(offset, &field));
+    }
+
+    GenImplementationGetters(struct_def, non_deprecated_fields, &code);
 
```

Comment:
It could be better to add a `Clone` method to the class `StructDef` and filter out deprecated fields from clone?
With this clone of type `StructDef` we don’t need to fix the rest of the code.

Another way, introduce an iterator for `SymbolTable<T>` which can filter out deprecated fields on the fly.
What do you think about it?

Reply:
We still need to know the original offsets of the non deprecated fields.

Reply:
Yes, this is a bit of a mess, all other code generators just have `if (field.deprecated) continue;` in every loop. An iterator would be very cool, but not necessarily up to the Dart generator to fix this.

Reply:
I cringed a bit looking at this code.  I had not written C++ in some time when I first wrote it, and it's a bit .. messy.  Sorry about that!